modules/tensor_mechanics/src/materials/LAROMANCEPartitionStressUpdateBase.C

```diff
@@ -0,0 +1,124 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "LAROMANCEPartitionStressUpdateBase.h"
+
+#include <libmesh/dense_matrix.h> // libMesh::cholesky_solve
+
+template <bool is_ad>
+InputParameters
+LAROMANCEPartitionStressUpdateBaseTempl<is_ad>::validParams()
+{
+  InputParameters params = LAROMANCEStressUpdateBaseTempl<is_ad>::validParams();
+  params.addClassDescription("LAROMANCE base class for partitioned reduced order models");
+
+  return params;
+}
+
+template <bool is_ad>
+LAROMANCEPartitionStressUpdateBaseTempl<is_ad>::LAROMANCEPartitionStressUpdateBaseTempl(
+    const InputParameters & parameters)
+  : LAROMANCEStressUpdateBaseTempl<is_ad>(parameters)
+{
+}
+
+template <bool is_ad>
+GenericReal<is_ad>
+LAROMANCEPartitionStressUpdateBaseTempl<is_ad>::computeSecondPartitionWeight()
+{
+  // extract and scale only the relevant inputs on which the GPR was trained (ignore
+  // _old_strain_input_index)
+  std::vector<GenericReal<is_ad>> scaled_input_values(_input_values.size() - 1);
+  unsigned int inc = 0;
+  for (unsigned int i = 0; i < _input_values.size(); i++)
```

Comment:
We tend to use

```suggestion
  for (const auto i: index_range(_input_values))
```

in new code these days. (`#include "libmesh/int_range.h"`)

Reply:
Sure we do that plenty. Here you seem to need the index though.

Reply:
Yeah, totally. Just curious. Is this a fix it now or fix it later suggestion here?

Reply:
Up to you. I can approve this now.